broker/src/test/java/io/zeebe/broker/exporter/stream/ExporterDirectorTest.java

```diff
@@ -386,6 +387,64 @@ public void shouldExecuteScheduledTask() throws Exception {
     assertThat(timerTriggerLatch.await(5, TimeUnit.SECONDS)).isTrue();
   }
 
+  @Test
+  public void shouldExecuteScheduledCancellableTask() throws InterruptedException {
+    // given
+    final CountDownLatch timerTriggerLatch = new CountDownLatch(1);
+    final CountDownLatch timerScheduledLatch = new CountDownLatch(1);
+    final Duration delay = Duration.ofSeconds(10);
+
+    final ControlledTestExporter exporter = exporters.get(0);
+    exporter.onExport(
+        r -> {
+          exporter.getController().scheduleCancellableTask(delay, timerTriggerLatch::countDown);
+          timerScheduledLatch.countDown();
+        });
+
+    // when
+    startExporterDirector(exporterDescriptors);
+
+    writeEvent();
+
+    timerScheduledLatch.await(5, TimeUnit.SECONDS);
+
+    rule.getClock().addTime(delay);
+
+    // then
+    assertThat(timerTriggerLatch.await(5, TimeUnit.SECONDS)).isTrue();
+  }
+
+  @Test
+  public void shouldCancelScheduledCancellableTask() throws InterruptedException {
+    // given
+    final CountDownLatch timerScheduledLatch = new CountDownLatch(1);
+    final Duration delay = Duration.ofSeconds(10);
+
+    final var exporter = exporters.get(0);
+    exporter.onExport(
+        r -> {
+          final var scheduledTask =
+              exporter
+                  .getController()
+                  .scheduleCancellableTask(
+                      delay, () -> fail("Must not execute this block. Task must be cancelled."));
+          scheduledTask.cancel();
+          timerScheduledLatch.countDown();
+        });
+
+    // when
+    startExporterDirector(exporterDescriptors);
+
+    writeEvent();
+
+    timerScheduledLatch.await(5, TimeUnit.SECONDS);
+
+    rule.getClock().addTime(delay);
+
+    // then
+    assertThat(exporter.getExportedRecords()).hasSize(1);
```

Comment:
Here, we want to verify if the task was actually cancelled. This assert does not ensure that the task was not executed. I I would suggest something similar to following:
```
startExporterDirector(exporterDescriptors);
var taskToCancel = exporter.getController().scheduleCancellableTask(delay, // set some value);
var taskToExecute = exporter.getController().scheduleCancellableTask(delay,   () -> latch.countdown());
taskToCancel.cancel();
// increase the clock
rule.getClock().addTime(delay);
// wait until the second task is executed
latch.await(..);
// Verify the cancelled task is not executed - that is the value is not set


```

Reply:
Done, considering: https://github.com/zeebe-io/zeebe/pull/5843#discussion_r530321724